### PR TITLE
DOP-3554

### PR DESCRIPTION
**Pull request by seungpark**
### Stories/Links:

DOP-3554

### Current Behavior:

[Atlas release notes](https://www.mongodb.com/docs/atlas/release-notes/)
[atlas cli with merged toc not showing active toc on load](https://www.mongodb.com/docs/atlas/cli/upcoming/)

### Staging Links:

[Staged Atlas release notes](https://docs-mongodbcom-integration.corp.mongodb.com/37d0cb05224bd2be9b52d42fc94af7dbfbf21f5a/master/cloud-docs/seung.park/DOP-3554/release-notes/)
[staged atlas cli](https://docs-mongodbcom-integration.corp.mongodb.com/37d0cb05224bd2be9b52d42fc94af7dbfbf21f5a/master/atlas-cli/seung.park/DOP-3554/atlas-cli-getting-started/)

### Notes:
- in order to reference the DOM element, had to wrap each SideNavItem in an empty div in order to set a DOM ref.
image 
otherwise, the other solution is to
1 use ref with SideNavItem since [it is a forwardRef component](https://github.com/mongodb/leafygreen-ui/blob/%40leafygreen-ui/side-nav%4010.0.1/packages/side-nav/src/SideNavItem/SideNavItem.tsx#L44-L47)
and 2 update Link to be a forwardRef component in order for the `as={Link}` portion to work well with the forwardref

TLDR; would it be better to use a singular forwardref wrapper component on the SideNavItem VERSUS making sure each SideNavItem component can be a forwardref component

### CODE REVIEW

File: src/components/Sidenav/Toctree.js [hunk 1]
```diff
@@ -6,7 +6,13 @@ const Toctree = ({ handleClick, slug, toctree: { children } }) => {
   return (
     <>
       {children.map((c) => (
-        <TOCNode key={c.slug || c.url || c.options?.urls} activeSection={slug} handleClick={handleClick} node={c} />
+        <TOCNode
+          key={c.slug || c.url || c.options?.urls}
+          activeSection={slug}
+          handleClick={handleClick}
+          node={c}
+          currentSlug={slug}
```
Comment (rayangler) [inline src/components/Sidenav/Toctree.js:14]: What is the difference between `currentSlug` and `activeSection`?

File: src/components/Sidenav/TOCNode.js [hunk 2]
```diff
@@ -76,6 +68,18 @@ const TOCNode = ({ activeSection, handleClick, level = BASE_NODE_LEVEL, node, pa
...
   const isTocIcon = !!(options.tocicon === 'sync');
   const [isOpen, setIsOpen] = useState(isActive);
+  const tocNodeRef = useRef(null);
+
+  // effect of scrolling toc node into view on load
+  const { completedFetch } = useContext(NavigationContext);
+  const isScrolled = useRef(false);
+  useEffect(() => {
+    if (isScrolled.current) return;
+    if (completedFetch && tocNodeRef.current && currentSlug === slug) {
```
Comment (rayangler) [inline src/components/Sidenav/TOCNode.js:78]: `currentSlug === slug` looks very similar to the logic represented by the `isActive` variable. Is that something that can be reused or is the use case different?

**Review by rayangler — COMMENTED** [review]
This functionality is so cool! Just a few things I wanted to ask about

File: src/components/Sidenav/TOCNode.js (hunk 2 above)
Comment (seungpark) [reply src/components/Sidenav/TOCNode.js:78]: ah great catch @rayangler ! removed this `currentSlug` variable and consolidated with existing`isSelectedTocNode` function.

File: src/components/Sidenav/Toctree.js (hunk 1 above)
Comment (seungpark) [reply src/components/Sidenav/Toctree.js:14]: nada. great catch :p

File: src/components/Sidenav/TOCNode.js [hunk 3]
```diff
@@ -115,12 +119,7 @@ const TOCNode = ({ activeSection, handleClick, level = BASE_NODE_LEVEL, node, pa
 
     if (isDrawer && hasChildren) {
       return (
-        <Wrapper
-          activeVersions={activeVersions}
-          hasVersions={hasVersions}
-          project={options.project}
-          versions={options.versions}
-        >
+        <Box ref={tocNodeRef} className={cx({ [wrapperStyle]: true }, { [wrapperPadding]: hasVersions })}>
```
[the remark this answers is not in the document]
Comment (seungpark) [reply src/components/Sidenav/TOCNode.js:122]: good catch! done

**Review by rayangler — APPROVED** [review]

File: src/components/Sidenav/TOCNode.js (hunk 2 above)
Comment (seungpark) [reply src/components/Sidenav/TOCNode.js]: was hesitant to use the component scoped const but included in dependency array of useEffect and same result 👍
